tests/vegalite/v5/test_api.py

```diff
@@ -630,36 +635,68 @@ def test_when_multiple_fields():
     with pytest.raises(TypeError, match=chain_mixed_msg):
         when.then("field_1:Q").when(Genre="pop")
 
+    chained_when = when.then(alt.value(5)).when(
+        alt.selection_point(fields=["b"]) | brush, empty=False, b=63812
+    )
+
+    chain_then_msg = re.compile(
+        r"Chained.+mixed.+field.+min\(foo\):Q.+'aggregate': 'min', 'field': 'foo', 'type': 'quantitative'",
+        flags=re.DOTALL,
+    )
+
+    with pytest.raises(TypeError, match=chain_then_msg):
+        chained_when.then("min(foo):Q")
+
     chain_otherwise_msg = re.compile(
         r"Chained.+mixed.+field.+AggregatedFieldDef.+'this_field_here'",
         flags=re.DOTALL,
     )
     with pytest.raises(TypeError, match=chain_otherwise_msg):
-        when.then(alt.value(5)).when(
-            alt.selection_point(fields=["b"]) | brush, empty=False, b=63812
-        ).then("min(foo):Q").otherwise(
+        chained_when.then(alt.value(2)).otherwise(
             alt.AggregatedFieldDef(
                 "argmax", field="field_9", **{"as": "this_field_here"}
             )
         )
 
 
 def test_when_typing(cars) -> None:
-    color = (
-        alt.when(alt.datum.Weight_in_lbs >= 3500)
-        .then(alt.value("black"))
-        .otherwise(alt.value("white"))
-    )
-    source = cars
-    chart = (  # noqa: F841
-        alt.Chart(source)
-        .mark_rect()
-        .encode(
-            x=alt.X("Cylinders:N").axis(labelColor=color),
-            y=alt.Y("Origin:N", axis=alt.Axis(tickColor=color)),
-            color=color,
+    chart = alt.Chart(cars).mark_rect()
+    predicate = alt.datum.Weight_in_lbs >= 3500
+    statement = alt.value("black")
+    default = alt.value("white")
+
+    then: alt.Then[_Conditions] = alt.when(predicate).then(statement)
+    otherwise: _Conditional[_Conditions] = then.otherwise(default)
+    condition: Map = alt.condition(predicate, statement, default)
+
+    # NOTE: both `condition()` and `when-then-otherwise` are allowed in these three locations
+    chart.encode(
+        color=condition,
+        x=alt.X("Cylinders:N").axis(labelColor=condition),
+        y=alt.Y("Origin:N", axis=alt.Axis(tickColor=condition)),
+    ).to_dict()
+
+    chart.encode(
+        color=otherwise,
+        x=alt.X("Cylinders:N").axis(labelColor=otherwise),
+        y=alt.Y("Origin:N", axis=alt.Axis(tickColor=otherwise)),
+    ).to_dict()
+
+    with pytest.raises(SchemaValidationError):
+        # NOTE: `when-then` is allowed as an encoding, but not as a `ConditionalAxisProperty`
+        # The latter fails validation since it does not have a default `value`
+        chart.encode(
+            color=then,
+            x=alt.X("Cylinders:N").axis(labelColor=then),  # type: ignore[call-overload]
+            y=alt.Y("Origin:N", axis=alt.Axis(labelColor=then)),  # type: ignore[arg-type]
```

Comment:
I was surprised that the current schema doesn't allow this.

Conditional encodings can omit the default, but conditional properties must define one.
I find this strange as all the conditional properties I tested accept `value=None` - so maybe this support could be added in `vega-lite`?